dbms/src/Interpreters/Context.h

```diff
@@ -239,6 +243,7 @@ class Context
 
     /// Get query for the CREATE table.
     ASTPtr getCreateQuery(const String & database_name, const String & table_name) const;
+    ASTPtr getCreateExternalQuery(const String & table_name) const;
```

Comment:
The name looks strange. This is because we have a mess with "temporary" and "external" tables.
Ok for now, but I would like to split "temporary" and "external" tables.